components/TextFields/src/MDCTextInputController.h

```diff
@@ -0,0 +1,200 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MDCTextFieldPositioningDelegate.h"
+
+@protocol MDCTextInput;
+@protocol MDCTextInputCharacterCounter;
+
+/**
+ Presentation styles for a text input. The style determines specific aspects of the text
+ input such as sizing, placeholder placement and behavior, layout, etc.
+ */
+typedef NS_ENUM(NSUInteger, MDCTextInputPresentationStyle) {
```

Comment:
Yes there is: app uses floating or default but then has a 'send me a message' view that needs full width.

I'm proposing that this be encapsulated into classes of controller instead of presentation styles in an enum. So you'd have FullWidthController class and FloatingPlaceholderController class so if you wanted to add EverythingsComingUpMillhouse style it would just be a controller class too.